Chirp/Networking/NetworkService.swift

```diff
@@ -77,5 +84,57 @@ struct NetworkService {
         .eraseToAnyPublisher()
     }
     
+    func createUser(request: CreateUserRequest) -> AnyPublisher<User, Error> {
+        Deferred {
+            Future { promise in
+                db.collection(Table.users.rawValue).document(request.userID).setData([
+                    Key.name.rawValue: request.name,
+                    Key.phoneNumber.rawValue: request.phoneNumber,
+                    Key.profilePicture.rawValue: request.profilePicture ?? ""
+                ]) { error in
+                    if let error = error {
+                        promise(.failure(error))
+                    }
+                    
+                    let user = User(id: request.userID,
+                                    name: request.name,
+                                    phoneNumber: request.phoneNumber,
+                                    profilePicture: request.profilePicture)
+                    promise(.success(user))
+                }
+            }
+        }
+        .receive(on: DispatchQueue.main)
+        .eraseToAnyPublisher()
+    }
+    
+    func uploadProfileImage(request: UploadProfileImageRequest) -> AnyPublisher<URL, Error> {
+        Deferred {
+            Future { promise in
+                
+                let storageRef = Storage.storage().reference()
+                let imagesCollectionRef = storageRef.child(Collection.images.rawValue)
+                let fileName = "profileImage_\(request.userID)_\(request.timestamp).\(request.imageExtension)"
+                let imageRef = imagesCollectionRef.child(fileName)
+                
+                imageRef.putData(request.imageData) { metadata, error in
+                    if let error = error {
+                        promise(.failure(error))
```

Comment:
Same feedback here.
Add return after the promise so that it does not execute the next line